source/common/upstream/outlier_detection_impl.cc

```diff
@@ -281,5 +391,24 @@ int EventLoggerImpl::secsSinceLastAction(const Optional<SystemTime>& lastActionT
   return -1;
 }
 
+SuccessRateAccumulatorBucket* SuccessRateAccumulator::updateCurrentWriter() {
+  // Right now current is being written to and backup is not. Flush the backup and swap.
+  backup_success_rate_bucket_->success_request_counter_ = 0;
+  backup_success_rate_bucket_->total_request_counter_ = 0;
+
+  current_success_rate_bucket_.swap(backup_success_rate_bucket_);
+
+  return current_success_rate_bucket_.get();
+}
+
+Optional<double> SuccessRateAccumulator::getSuccessRate(uint64_t success_rate_request_volume) {
+  if (backup_success_rate_bucket_->total_request_counter_ < success_rate_request_volume) {
+    return Optional<double>();
+  }
+
+  return Optional<double>(backup_success_rate_bucket_->success_request_counter_ * 100 /
```

Comment:
you are losing precision here, should probably sr_counter_ * 100.0 / total_coutner_